### PR TITLE
Remove model.AppError from searchlayer/channel_layer.go

### LINKED ISSUE #16416: Remove model.AppError from searchlayer/channel_layer.go
We are in the process of removing `model.AppError` completely from the `store` package. We have removed it from the `sqlstore` package, but there are still a couple left in some other packages under `store`.

This ticket is about removing `model.AppError` from searchlayer/channel_layer.go

The way to do that would be to use the different error types in store/errors.go if a particular error falls in that category. And if it's just a generic error, then just return an `errors.Wrapf` wrapping the original error.

If you have questions about the ticket or you need any help, feel free to contact `agniva.de.sarker` in https://community.mattermost.com/

----------------------------------

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

### PR DESCRIPTION
**Pull request by msal4**
#### Summary

Removes all references to model.AppError.

I've also noticed that the code editor complains about some parameter names (e.g. method parameter `userId` should be `userID`). I've left them untouched as they're out of scope for this pull request.

#### Ticket Link

Fixes #16416

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/searchlayer/channel_layer.go [hunk 1]
```diff
@@ -130,15 +130,15 @@ func (c *SearchChannelStore) SaveDirectChannel(directchannel *model.Channel, mem
 
 func (c *SearchChannelStore) AutocompleteInTeam(teamId string, term string, includeDeleted bool) (*model.ChannelList, error) {
 	var channelList *model.ChannelList
-	var appErr *model.AppError
+	var err error
 	var nErr error
```
Comment (agnivade) [inline store/searchlayer/channel_layer.go:134]: We can now remove this and just replace `nErr` by `err`.

File: store/searchlayer/channel_layer.go [hunk 2]
```diff
@@ -151,18 +151,18 @@ func (c *SearchChannelStore) AutocompleteInTeam(teamId string, term string, incl
 		mlog.Debug("Using database search because no other search engine is available")
 		channelList, nErr = c.ChannelStore.AutocompleteInTeam(teamId, term, includeDeleted)
 		if nErr != nil {
-			return nil, model.NewAppError("AutocompleteInTeam", "app.channel.search.app_error", nil, nErr.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrapf(err, utils.T("app.channel.search.app_error"))
```
Comment (agnivade) [inline store/searchlayer/channel_layer.go:154]: The translations are only done at the top layer when the error message is returned to the client. We don't need to do it for internal errors like these. I'd recommend: `errors.Wrapf(err, "Failed to autocomplete channels in team")`

File: store/searchlayer/channel_layer.go [hunk 3]
```diff
@@ -172,7 +172,7 @@ func (c *SearchChannelStore) searchAutocompleteChannels(engine searchengine.Sear
 	if len(channelIds) > 0 {
 		channels, err := c.ChannelStore.GetChannelsByIds(channelIds, includeDeleted)
 		if err != nil {
-			return nil, model.NewAppError("searchAutocompleteChannels", "app.channel.get_channels_by_ids.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrapf(err, utils.T("app.channel.get_channels_by_ids.app_error"))
```
Comment (agnivade) [inline store/searchlayer/channel_layer.go:175]: Same as above.

**Review by agnivade — CHANGES_REQUESTED** [review]
Some minor changes are needed. Looks good otherwise.

- Why is an empty output file added to the PR?
- You might also need to run `make i18n-extract` and see if there are any changes.

File: store/searchlayer/channel_layer.go (hunk 2 above)
Comment (msal4) [reply store/searchlayer/channel_layer.go:154]: good to know. thanks

**Comment by msal4** [comment]
>     * Why is an empty output file added to the PR?

I think the output file was generated while running tests but im not sure, i've removed it now.

>     * You might also need to run make i18n-extract and see if there are any changes.
done

File: store/searchlayer/channel_layer.go [hunk 4]
```diff
@@ -172,7 +170,7 @@ func (c *SearchChannelStore) searchAutocompleteChannels(engine searchengine.Sear
 	if len(channelIds) > 0 {
 		channels, err := c.ChannelStore.GetChannelsByIds(channelIds, includeDeleted)
 		if err != nil {
-			return nil, model.NewAppError("searchAutocompleteChannels", "app.channel.get_channels_by_ids.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrapf(err, "Failed to autocomplete channels in team")
```
Comment (agnivade) [inline store/searchlayer/channel_layer.go:173]: The error message should be different here. Sorry I thought that was implied. We can have something like "Failed to get channels by ids"

**Review by agnivade — CHANGES_REQUESTED** [review]
Sorry, my apologies. Please revert the changes to en.json. I forgot that you need access to our EE codebase to run it properly. I verified that there are no changes needed to it.

File: store/searchlayer/channel_layer.go (hunk 4 above)
Comment (msal4) [reply store/searchlayer/channel_layer.go]: sorry, i made the necessary changes

**Review by agnivade — APPROVED** [review]
LGTM!

**Review by jespino — APPROVED** [review]
Looks good to me. Thanks @msal4! :dancing_women:

**Comment by jespino** [comment]
@agnivade I assigned @lindy65 for QA this, I guess some smoke testing around channel autocompletion should be enough. (I don't know if is not needed because is part of the AppError refactor.

**Comment by lindy65** [comment]
Thanks @jespino :)

Is there a Jira ticket for this PR @agnivade?  I usually find one on the GitHub issue but don't see one there either?

**Comment by agnivade** [comment]
Hi @lindy65! Here's the GitHub issue https://github.com/mattermost/mattermost-server/issues/16416.

**Review by lindy65 — APPROVED** [review]
Thanks @msal4 @agnivade - looks good to me 👍

**Pull request merged**